### PR TITLE
VSR restoreItemAction v2

### PR DESCRIPTION
**Pull request by eemcmullan**

### CODE REVIEW

File: internal/restore/volumesnapshotrestore_action.go [hunk 1]
```diff
@@ -71,19 +77,101 @@ func (p *VolumeSnapshotRestoreRestoreItemAction) Execute(input *velero.RestoreIt
...
+	return progress, nil
+}
+
+func (p *VolumeSnapshotRestoreRestoreItemActionV2) Cancel(operationID string, restore *v1.Restore) error {
+	return nil
+}
+
+func (p *VolumeSnapshotRestoreRestoreItemActionV2) AreAdditionalItemsReady(additionalItems []velero.ResourceIdentifier, restore *v1.Restore) (bool, error) {
+	return true, nil
+}
```
Comment (kaovilai) [inline internal/restore/volumesnapshotrestore_action.go:177]: ehh these two funcs are empty and doesn't do anything with the params

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: internal/restore/volumesnapshotrestore_action.go (hunk 1 above)
Comment (sseago) [reply internal/restore/volumesnapshotrestore_action.go:177]: @kaovilai Yes, they're needed to implement the interface. If canceling the action is possible, then some implementation could be added.

**Review by sseago — APPROVED** [review]

File: internal/restore/volumesnapshotrestore_action.go (hunk 1 above)
Comment (kaovilai) [reply internal/restore/volumesnapshotrestore_action.go]: Then comment as not implemented/

Comment (kaovilai) [reply internal/restore/volumesnapshotrestore_action.go]: `// empty funcs to satisfy interface`

File: internal/util/util.go [hunk 2]
```diff
@@ -375,7 +375,7 @@ func GetVolumeSnapshotRestoreWithStatusData(restoreName string, PVCName string,
 		}
 
 		if len(vsrList.Items) > 0 {
```
Comment (kaovilai) [inline internal/util/util.go:377]: If user have made a restore before with the same name, it is a possibility that a vsr might exists from a different restore of the same name. Should we be using restore  UID instead/in addition to the restore name?

It is common in testing for cu to delete restore and try again and may got the wrong VSR for status.

Comment (eemcmullan) [reply internal/util/util.go:377]: We will need a deleteItemAction plugin to handle deleting VSR when its restore is deleted. This is how we handle it for backup/VSB, which is outside of this plugin.

Comment (eemcmullan) [reply internal/util/util.go:377]: Going to add logic to the VSB deleteItemAction to also delete corresponding VSR to handle such case, so will fix outside of this PR

Comment (kaovilai) [reply internal/util/util.go:377]: I still think adding UID would be a good idea. Similar to ownerReferences which has UID as well.

**Review by kaovilai — APPROVED** [review]

**Pull request merged**